py-polars/tests/unit/datatypes/test_temporal.py

```diff
@@ -901,7 +901,9 @@ def test_groupby_dynamic_crossing_dst(rule: str, offset: timedelta) -> None:
         start_dt, end_dt, rule, time_zone="US/Central", eager=True
     )
     df = pl.DataFrame({"time": date_range, "value": range(len(date_range))})
-    result = df.groupby_dynamic("time", every=rule).agg(pl.col("value").mean())
+    result = df.groupby_dynamic("time", every=rule, start_by="datapoint").agg(
```

Comment:
same as in the Rust test - the default `start_by='window'` would actually act like `start_by='datapoint'` for `'w'`, so I'm setting `start_by='datapoint'` to preserve the behaviour of the test